frontend/src/lib/components/TaxonomicFilter/InfiniteList.tsx

```diff
@@ -0,0 +1,254 @@
+            listGroupType === TaxonomicFilterGroupType.EventProperties ||
+            listGroupType === TaxonomicFilterGroupType.PersonProperties
+        ) {
+            data = getKeyMapping(value.toString(), 'event')
+        } else if (listGroupType === TaxonomicFilterGroupType.Elements) {
+            data = getKeyMapping(value.toString(), 'element')
+        }
+
+        if (data) {
+            return (
+                <div style={{ width }}>
+                    <PropertyKeyTitle data={data} />
+                    {data.description ? <hr /> : null}
+                    <PropertyKeyDescription data={data} value={value.toString()} />
+
+                    {'volume_30_day' in item && (item.volume_30_day || 0) > 0 ? (
+                        <p>
+                            Seen <strong>{item.volume_30_day}</strong> times.{' '}
+                        </p>
+                    ) : null}
+                    {'query_usage_30_day' in item && (item.query_usage_30_day || 0) > 0 ? (
+                        <p>
+                            Used in <strong>{item.query_usage_30_day}</strong> queries.
+                        </p>
+                    ) : null}
+                </div>
+            )
+        }
+    }
+
+    return item.name ?? ''
+}
+
+const selectedItemHasPopup = (
+    item?: PropertyDefinition | CohortType,
+    listGroupType?: TaxonomicFilterGroupType,
+    group?: TaxonomicFilterGroup
+): boolean => {
+    return (
+        // NB: also update "renderItemPopup" above
+        !!item &&
+        !!group?.getValue(item) &&
+        (listGroupType === TaxonomicFilterGroupType.Actions ||
+            ((listGroupType === TaxonomicFilterGroupType.Elements ||
+                listGroupType === TaxonomicFilterGroupType.Events ||
+                listGroupType === TaxonomicFilterGroupType.EventProperties ||
+                listGroupType === TaxonomicFilterGroupType.PersonProperties) &&
+                !!getKeyMapping(
+                    group?.getValue(item),
+                    listGroupType === TaxonomicFilterGroupType.Elements ? 'element' : 'event'
+                )))
+    )
+}
+
+export function InfiniteList(): JSX.Element {
+    const { mouseInteractionsEnabled, activeTab, searchQuery, value, groupType } = useValues(taxonomicFilterLogic)
+    const { selectItem } = useActions(taxonomicFilterLogic)
+
+    const { isLoading, results, totalCount, index, listGroupType, group, selectedItem, selectedItemInView } = useValues(
+        infiniteListLogic
+    )
+    const { onRowsRendered, setIndex } = useActions(infiniteListLogic)
+
+    const isActiveTab = listGroupType === activeTab
+    const showEmptyState = totalCount === 0 && !isLoading
+
+    const [referenceElement, setReferenceElement] = useState<HTMLDivElement | null>(null)
+    const [popperElement, setPopperElement] = useState<HTMLDivElement | null>(null)
+
+    const { styles, attributes } = usePopper(referenceElement, popperElement, {
```

Comment:
Cool! Going forward will `react-popper` replace antd Tooltips?

Reply:
I hope so! Since `react-popper` itself is just a hook, we need to build the actual popup component and its interactions with the rest of the app ourselves. This can get tricky. For example I already had to include [some funky code](https://github.com/PostHog/posthog/pull/5189/files#diff-f697c1f22f4b126019f5e8ecaf42488faab025d47baf2fbc6dcc6f4debf3e17bR36-R52) to make sure that we can have multiple popups on the page at the same time, without interfering in each other's `onClickOutside` logic. There will surely be more stuff like this to come.